src/pveclib/vec_int16_ppc.h

```diff
@@ -615,6 +622,58 @@ vec_clzh (vui16_t vra)
   return (r);
 }
 
+/** \brief Vector Count Trailing Zeros Halfword.
+ *
+ *  Count the number of trailing '0' bits (0-16) within each halfword
+ *  element of a 128-bit vector.
+ *
+ *  For POWER9 (PowerISA 3.0B) or later use the Vector Count Trailing
+ *  Zeros Halfword instruction <B>vctzh</B>. Otherwise use a sequence of
+ *  pre ISA 3.0 VMX instructions.
+ *  SIMDized count Trailing zeros inspired by:
+ *
+ *  Warren, Henry S. Jr and <I>Hacker's Delight</I>, 2nd Edition,
+ *  Addison Wesley, 2013. Chapter 5 Counting Bits, Section 5-4.
+ *
+ *  |processor|Latency|Throughput|
+ *  |--------:|:-----:|:---------|
+ *  |power8   |  6-8  | 2/cycle  |
+ *  |power9   |   3   | 2/cycle  |
+ *
+ *  @param vra 128-bit vector treated as 8 x 16-bit short integer
+ *  (halfwords) elements.
+ *  @return 128-bit vector with the Trailng Zeros count for each
+ *  halfword element.
+ */
+static inline vui16_t
+vec_ctzh (vui16_t vra)
+{
+  vui16_t r;
+#ifdef _ARCH_PWR9
+#if defined (vec_cnttz) || defined (__clang__)
+  r = vec_cnttz (vra);
+#else
+  __asm__(
+      "vctzh %0,%1;"
+      : "=v" (r)
+      : "v" (vra)
+      : );
```

Comment:
subjective, but this could all fit on one line

Reply:
See response above